### PR TITLE
HDDS-9821. XceiverServerRatis SyncTimeoutRetry is overridden

### PR DESCRIPTION
**Pull request by ivandika3**
## What changes were proposed in this pull request?

In XceiverServerRatis#newRaftProperties, setSyncTimeoutRetry was set twice

First, it is set to `(int) nodeFailureTimeoutMs / dataSyncTimeout.toIntExact(TimeUnit.MILLISECONDS)` which by default equals to 300_000 ms / 10_000 ms  =  30 retries (each waiting for 10 seconds)

From the comment, the intention of setting a finite number of retries is:

"Even if the leader is not able to complete write calls within the timeout seconds, it should just fail the operation and trigger pipeline close. failing the writeStateMachine call with limited retries will ensure even the leader initiates a pipeline close if its not able to complete write in the timeout configured."

However, it was overridden in

```java 
int numSyncRetries = conf.getInt(
    OzoneConfigKeys.DFS_CONTAINER_RATIS_STATEMACHINEDATA_SYNC_RETRIES,
    OzoneConfigKeys.
        DFS_CONTAINER_RATIS_STATEMACHINEDATA_SYNC_RETRIES_DEFAULT);
RaftServerConfigKeys.Log.StateMachineData.setSyncTimeoutRetry(properties,
    numSyncRetries); 
```

Which set it to the default value -1 (retry indefinitely). 

This might cause the leader to never initiate a pipeline close when its writeStateMachine time out (e.g. due to I/O issue during write chunk).

I propose we use the calculation based on the node failure by default and drop the -1 default value configuration.

I also added related comments about state machine data cache for better understanding. Please let me know if the explanation can be improved.

This is also good avenue to re-evaluate the state machine data policy in Container State Machine.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9821

## How was this patch tested?

Existing tests. Only configurations change.

Clean run: https://github.com/ivandika3/ozone/actions/runs/7068898435

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java [hunk 1]
```diff
@@ -381,9 +381,6 @@ public final class OzoneConfigKeys {
   public static final String
       DFS_CONTAINER_RATIS_STATEMACHINEDATA_SYNC_RETRIES =
       ScmConfigKeys.DFS_CONTAINER_RATIS_STATEMACHINEDATA_SYNC_RETRIES;
-  public static final int
-      DFS_CONTAINER_RATIS_STATEMACHINEDATA_SYNC_RETRIES_DEFAULT =
-      ScmConfigKeys.DFS_CONTAINER_RATIS_STATEMACHINEDATA_SYNC_RETRIES_DEFAULT;
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:386]: `ScmConfigKeys.DFS_CONTAINER_RATIS_STATEMACHINEDATA_SYNC_RETRIES_DEFAULT` is now unused, can be removed.

Also, I think we should remove the default value for `dfs.container.ratis.statemachinedata.sync.retries` in `ozone-default.xml` (set it to `<value/>`), and also update the description to explain that the default value is calculated based on other properties.

https://github.com/apache/ozone/blob/74de0705c916bb877eae44310f43a0d7069cdef9/hadoop-hdds/common/src/main/resources/ozone-default.xml#L100-L105

Comment (ivandika3) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:386]: @adoroszlai Thank you for the review. I have removed the unused configuration and updated the ozone-default.xml with the description. Please let me know if the description can be improved further.

**Review by szetszwo — APPROVED** [review]
@ivandika3 , thanks a lot for working on this and also the well written code comments!

+1 the change looks good.

**Comment by szetszwo** [comment]
@adoroszlai , thank you also for reviewing this!

**Comment by ivandika3** [comment]
Thank you for the reviews @adoroszlai @szetszwo

**Pull request merged**